2-frontend/src/components/app/InvoicesList/tests/InvoicesList.test.jsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import InvoicesList from "../InvoicesList";
+import Pagination from "../../../visual/Pagination/Pagination";
+import Filter from "../../../visual/Filter/Filter";
+import Table from "../../../visual/Table/Table";
+import getInvoices from "../../../../services/getInvoices";
+
+/**
+ * MOCKS **************************
+ */
+
+jest.mock("../../../../services/getInvoices");
+
+const mockInvoices = [
+  {id: "mockInvoice1"}, 
+  {id: "mockInvoice2"}, 
+  {id: "mockInvoice3"}
+];
+getInvoices.mockImplementation(() => ({
+  items: mockInvoices,
+  total: 10
+}));
+
+/**
+ * ********************************
+ */
+
+/**
+ * DEFAULT STATES **************************
+ */
+
+const defaultInvoices = {
+  items: [],
+  total: 0,
+  requested: false
+};
+
+const defaultPageOpts = {
+  page: 0,
+  offset: 7
+};
+
+const defaultFilter = {
+  by: null,
+  val: ""
+};
+
+ /**
+  * ****************************************
+  */
+
+const getShallowWrapper = () => (
+  shallow(
+    <InvoicesList />
+  )
+);
+
+const getRequestInvoicesBtn = wrapper => (
+  wrapper.find("#getInvoicesBtn").at(0)
+);
+
+const getPagination = wrapper => (
+  wrapper.find(Pagination).at(0)
+);
+
+const getFilter = wrapper => (
+  wrapper.find(Filter).at(0)
+);
+
+const getTable = wrapper => (
+  wrapper.find(Table).at(0)
+);
+
+const assertRequestInvoices = (wrapper, pageOpts, filter) => {
+  const btn = getRequestInvoicesBtn(wrapper);
+  btn.simulate("click");
+  expect(getInvoices).toHaveBeenCalledTimes(1);
+  expect(getInvoices).toHaveBeenCalledWith(pageOpts, filter); 
+};
+
+describe("InvoicesList", () => {
```

Comment:
Hey @BernardoMG, thanks for your review/feedback 😄 
Usually, when developing new React components, I try to make them the smallest and well-scoped possible. Not only because I think it's clearer when we read the code but also to make unit tests easier. I think that when we use `shallow` testing we are able to focus more on the specific component that we are targeting, without being influenced by other components behaviour.
Not to say that `mount` shouldn't be used, as it provides a closer environment to the one where the component is going to be used. Probably in this task, the `InvoicesList` would be a good candidate to also be tested using `mount`, since it's a component that aggregates other components, so that we can check how it integrates with them.

Yes, I'm familiar with `snapshot testing` and I already use it before, in some of my components. However, I think that after a while, snapshot testing contributes to `if-failed-then-update` kind of approach, since they will break on almost every change that you make in the components. I'm not their biggest fan because of their fragility, but I agree that they add another layer to the testing coverage and they can be useful in more static/visual components where you want to maintain the generated DOM consistent (e.g: the components in the `visual` folder in this task).

A good approach would also be to include `end-to-end` tests by using some automation framework, as `selenium` or `cypress` to cover some test cases in the final environment where the app will run, the browser.